projects/observability/src/shared/components/cartesian/d3/axis/cartesian-axis.ts

```diff
@@ -79,17 +78,87 @@ export class CartesianAxis<TData = {}> {
     if (!this.configuration.tickLabels) {
       selection.selectAll('.tick text').remove();
     }
+
+    if (this.configuration.location === AxisLocation.Bottom) {
+      const maxTextTickTextLength = this.getMaxTickTextLength(selection);
+      const isLabelRotated = this.rotateAxisTicks(selection, maxTextTickTextLength);
+      this.removeOverflowedTicks(selection, maxTextTickTextLength, isLabelRotated);
+    } else {
+      this.maybeTruncateAxisTicks(selection);
+    }
+  }
+
+  private getMaxTickTextLength(axisSvgSelection: Selection<SVGGElement, unknown, null, undefined>): number {
+    const ticksSelection = axisSvgSelection.selectAll('text');
+
+    const allElementLength: number[] = [];
+
+    ticksSelection.each((_, index, nodes) =>
+      allElementLength.push(this.svgUtilService.getElementTextLength(nodes[index] as SVGTextElement))
+    );
+
+    return Math.max(...allElementLength);
   }
 
   private maybeTruncateAxisTicks(axisSvgSelection: Selection<SVGGElement, unknown, null, undefined>): void {
     const ticksSelection = axisSvgSelection.selectAll('text');
     const tickBandwidth = Math.abs(this.scale.getRangeEnd() - this.scale.getRangeStart()) / ticksSelection.size();
 
-    ticksSelection.each((_datum, index, nodes) =>
+    ticksSelection.each((_, index, nodes) =>
       this.svgUtilService.truncateText(nodes[index] as SVGTextElement, tickBandwidth)
     );
   }
 
+  private removeOverflowedTicks(
+    axisSvgSelection: Selection<SVGGElement, unknown, null, undefined>,
+    maxTextTickTextLength: number,
+    isLabelRotated: boolean
+  ): void {
+    axisSvgSelection.selectAll('.tick').each((_, i, n) => {
+      const currentTick = select(n[i]);
+
+      const getTickTranslateXValue = (tick: Selection<BaseType, unknown, null, undefined>): number => {
+        const translateXValue = tick
+          .attr('transform')
+          .replace(/.*\(|\).*/g, '')
+          .split(',')[0];
+
+        return parseInt(translateXValue);
+      };
+
+      const currentTickPosition = getTickTranslateXValue(currentTick);
+
+      if (
+        currentTickPosition < maxTextTickTextLength / 2 ||
```

Comment:
Rather than inlining a complex boolean statement like this, assigning values like `const isTick<Condition> = <condition predicate>` allows the code to comment itself. Either that, or a comment here would help explain what we're checking.

Reply:
fixed